### DISCUSSION TITLE
How to dynamically pass modifier to x-anchor?

### DISCUSSION THREAD

**Discussion by IHIutch** [alpinejs/alpine#3953] [1. Help]
I'm building a library using Alpine.js and creating a custom tooltip directive. I want users to be able to dynamically specify the position of the tooltip. Is there a way I pass a data attribute, or some such, to my directive in a way that lets me dynamically control the position modifier of `x-anchor`?

**Comment by ekwoka** [comment]
Yes.

with x-bind you should be able to provide an object that could choose the attributes to add.

Or at least should. There are the occasional unexpected behavior with dynamic binds.

**Reply by IHIutch** [reply]
Ah, okay, thanks. This seems to work:

```
'x-bind'() {
      return {
        [`x-anchor.${this.position}.offset.5`]() {
          return this.triggerEl;
        },
      };
    },
